src/SSHDebugPS/CommandRunner.cs

```diff
@@ -9,304 +9,304 @@
                 {
                     _hasExited = true;
-                    Closed?.Invoke(this, _localProcess.ExitCode);
+                    Closed?.Invoke(this, _process.ExitCode);
                 }
             }
         }
-    }
-
-    /// <summary>
-    /// Launches a local command that sends output when a newline is received.
-    /// </summary>
-    internal class LocalBufferedCommandRunner : ICommandRunner
-    {
-        private System.Diagnostics.Process _localProcess;
-        private StreamWriter _stdoutWriter;
-        private ProcessStartInfo processStartInfo;
-
-        private bool _isClosed = false;
 
-        public LocalBufferedCommandRunner(string command, string arguments)
+        protected virtual void OnErrorReceived(string error)
         {
-            processStartInfo = new ProcessStartInfo(command, arguments);
-            processStartInfo.RedirectStandardError = true;
-            processStartInfo.RedirectStandardInput = true;
-            processStartInfo.RedirectStandardOutput = true;
-            processStartInfo.UseShellExecute = false;
-            processStartInfo.CreateNoWindow = true;
-
-            _localProcess = System.Diagnostics.Process.Start(processStartInfo);
-            _stdoutWriter = new StreamWriter(_localProcess.StandardInput.BaseStream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), 4096, leaveOpen: false);
-            _localProcess.OutputDataReceived += OnProcessOutput;
-            _localProcess.ErrorDataReceived += OnErrorOutput;
-            _localProcess.Exited += OnProcessExited;
-
-            _localProcess.BeginOutputReadLine();
-            _localProcess.BeginErrorReadLine();
-            _localProcess.EnableRaisingEvents = true;
+            ErrorOccured?.Invoke(this, new ErrorOccuredEventArgs(error));
         }
 
-        public event EventHandler<string> OutputReceived;
-        public event EventHandler<int> Closed;
-        public event EventHandler<ErrorOccuredEventArgs> ErrorOccured;
+        protected virtual void OnOutputReceived(string line)
+        {
+            OutputReceived?.Invoke(this, line);
+        }
 
-        public void Dispose()
+        protected virtual void CreateProcessStartInfo(string command, string commandArgs)
         {
-            if (!_isClosed)
+            if (_process != null)
             {
-                _isClosed = true;
-                if (_localProcess != null)
+                if (!_process.HasExited)
                 {
-                    _localProcess.Close();
-                    _localProcess = null;
+                    Debug.Fail("CreateProcessStartInfo called when there is already a process running.");
                 }
-
-                if (_stdoutWriter != null)
+                else
                 {
-                    _stdoutWriter.Close();
-                    _stdoutWriter = null;
+                    CleanUpProcess();
                 }
             }
```

Comment:
If you make this method non-virtual, this block should no longer be necessary, as it's only ever called from the constructor, so there's no way the process could exist yet.